src/dist/mod.rs

```diff
@@ -208,6 +208,17 @@ impl fmt::Display for PartialVersion {
 impl FromStr for PartialVersion {
     type Err = anyhow::Error;
     fn from_str(ver: &str) -> Result<Self> {
+        // `semver::Comparator::from_str` supports an optional operator
+        // (e.g. `=`, `>`, `>=`, `<`, `<=`, `~`, `^`, `*`) before the
+        // partial version, so we should exclude that case first.
+        if let Some(ch) = ver.chars().nth(0) {
+            if !ch.is_ascii_digit() {
+                return Err(anyhow!(
+                    "expected ASCII digit at the beginning of `{ver}`, found `{ch}`"
+                )
+                .context("error parsing `PartialVersion`"));
```

Comment:
In my mind `context()` is usually used to give more context in error messages that you don't control (from other libraries). I'm not sure the separation between the error message and the context makes sense here?

Reply:
@djc Oh, that's just a tiny trick I use to make the output more consistent:

```rust
use anyhow::anyhow;
use anyhow::Context;
use std::str::FromStr;

fn main() {
    foo().unwrap();
    /*
    thread 'main' panicked at src/main.rs:5:11:
    called `Result::unwrap()` on an `Err` value: error parsing `PartialVersion`
    
    Caused by:
        expected ASCII digit at the beginning of...
    */
    bar().unwrap();
    /*
    thread 'main' panicked at src/main.rs:14:11:
    called `Result::unwrap()` on an `Err` value: error parsing `PartialVersion`
    
    Caused by:
        invalid digit found in string
    */
}

fn foo() -> anyhow::Result<()> {
    Err(
        anyhow!("expected ASCII digit at the beginning of...")
            .context("error parsing `PartialVersion`"),
    )
}

fn bar() -> anyhow::Result<()> {
    i32::from_str("bar").context("error parsing `PartialVersion`")?;
    Ok(())
}
```

Ideally if we had `try_block` I'll add the `.context()` after the `try{}`, however...